Weather/WeatherService.swift

```diff
@@ -37,58 +142,500 @@ struct WeatherData: Codable {
+    let obs_name : String?
+    let pressure_desc : String?
+    let pressure_tend : Int?
+    let qualifier : String?
+    let rh : Int?
+    let uv_desc : String?
+    let uv_index : Int?
+    let wdir : Int?
+    let wdir_cardinal : String?
+    let wx_icon : Int?
+    let wx_phrase : String?
+    let icon_extd : Int?
+
+    enum CodingKeys: String, CodingKey {
+
+        case classa = "class"
+        case valid_time_gmt = "valid_time_gmt"
+        case imperial = "imperial"
+        case metric = "metric"
+        case obs_id = "obs_id"
+        case obs_name = "obs_name"
+        case pressure_desc = "pressure_desc"
+        case pressure_tend = "pressure_tend"
+        case qualifier = "qualifier"
+        case rh = "rh"
+        case uv_desc = "uv_desc"
+        case uv_index = "uv_index"
+        case wdir = "wdir"
+        case wdir_cardinal = "wdir_cardinal"
+        case wx_icon = "wx_icon"
+        case wx_phrase = "wx_phrase"
+        case icon_extd = ""
+    }
+
+}
+public var name = "Montrèal"
+public var DisplayIcon: Int?
+private var Icon: String?
+
+internal class WeatherService: NSObject, CLLocationManagerDelegate {
+    let locationManager = CLLocationManager()
+    override init() {
+        super.init()
+        locationManager.delegate = self
+    }
+    func currentConditions(result:@escaping  (WeatherData?) -> Void) {
+        let Days_CodeName = ["Yesterday", "Today", "Tommorow"]
+        print(Info)
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.startUpdatingLocation()
+        guard let latitude = locationManager.location?.coordinate.latitude,
+              let longitude = locationManager.location?.coordinate.longitude else {
+            print("Unable to get location")
             return
         }
-        let request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData, timeoutInterval: 20)
-        let session = URLSession.shared.dataTask(with: request) { [result] data, response, error in
-            print("[WeatherService]: \((response as? HTTPURLResponse)?.statusCode ?? -999)")
-            guard error == nil, let data = data else {
-                let name: String = Preferences[.city_name]
-                let data = WeatherData(
-                    metadata: .init(error: nil, code: -999),
-                    weather: Weather(
-                        name: name,
-                        temp: -999,
-                        icon: NSImage.touchBarSearchTemplateName,
-                        description: "Unknown information"
-                    )
-                )
-                result(data)
+        locationManager.stopUpdatingLocation()
+        var description = " "
+        var temp = "0"
+        var temp_F = "0"
+        let url_for_maps = URL(string:  "https://dev.virtualearth.net/REST/v1/Locations/\(latitude),\(longitude)?includeNeighborhood=1&include=ciso2&o=json&key=Amn3nivKYow4ej0mliZVLFonh7W6ZBOlQz8FVrH0AJbK6_wViio1P8mrDODAgs6U")!
```

Comment:
Get's city, neighbourhood, address and other info based on coordinates using bing maps api.